cmd/crank/beta/xpkg/init.go

```diff
@@ -116,14 +133,132 @@ func (c *initCmd) Run(k *kong.Context, logger logging.Logger) error {
 }
 
+// handleNotes templates and prints the NOTES.txt file in the template
+// repository, if it exists.
+func handleNotes(k *kong.Context, logger logging.Logger, fs billy.Filesystem, dir, name string) error {
+	notesFile := dir + "/" + notes
+	f, err := fs.Stat(notesFile)
+	switch {
+	case os.IsNotExist(err):
+		// no NOTES.txt file, skip
+		logger.Debug("No NOTES.txt found, skipping")
+		return nil
+	case err != nil:
+		return errors.Wrapf(err, "failed to stat notes file %s", notesFile)
+	case f.IsDir():
+		return errors.Errorf("%s is not a file", notesFile)
+	}
+
+	filler := nameFiller{Name: name}
+	nfi, err := fs.Open(notesFile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to open notes file %s", notesFile)
+	}
+	ntRaw, err := io.ReadAll(nfi)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read notes file %s", notesFile)
+	}
+	nt, err := template.New("notes").Parse(string(ntRaw))
+	if err != nil {
+		return errors.Wrapf(err, "failed to parse notes file %s", notesFile)
+	}
+	err = nt.Execute(k.Stdout, filler)
+	if err != nil {
+		return errors.Wrapf(err, "failed to template notes file %s", notesFile)
+	}
+	// add a newline after the notes
+	_, err = fmt.Fprintln(k.Stdout)
+	if err != nil {
+		return errors.Wrap(err, "failed to write to stdout")
+	}
+	return nil
+}
+
+type nameFiller struct {
+	Name string
+}
+
+// handleInitScript runs the init.sh script in the template repository, if it
+// exists.
+func handleInitScript(k *kong.Context, logger logging.Logger, fs billy.Filesystem, dir, name string) error {
+	scriptFile := dir + "/" + initScript
+	f, err := fs.Stat(scriptFile)
+	switch {
+	case os.IsNotExist(err):
+		// no init.sh file, skip
+		logger.Debug("No init.sh found, skipping")
+		return nil
+	case err != nil:
+		return errors.Wrapf(err, "failed to stat init.sh file %s", scriptFile)
+	case f.IsDir():
+		return errors.Errorf("%s is not a file", scriptFile)
+	}
+
+	// read and print the script
+	sfi, err := fs.Open(scriptFile)
+	if err != nil {
+		return errors.Wrapf(err, "failed to open init.sh file %s", scriptFile)
+	}
+	sfRaw, err := io.ReadAll(sfi)
+	if err != nil {
+		return errors.Wrapf(err, "failed to read init.sh file %s", scriptFile)
+	}
+	// TODO(lsviben): maybe print the script in different color to make it stand out?
+	_, err = fmt.Fprintf(k.Stdout, "\nFound init script:\n%s\n", sfRaw)
+	if err != nil {
+		return errors.Wrap(err, "failed to write to stdout")
+	}
+
+	// ask the user if they want to run the script
+	yes, err := promptYesNo(k, "\nDo you want to run it?")
+	if err != nil {
+		return errors.Wrap(err, "failed to prompt user")
+	}
+	if !yes {
+		return nil
+	}
+	// execute the script
+	cmd := exec.Command(scriptFile, name, dir) //nolint:gosec // variable is provided by the user
+	cmd.Stdout = k.Stdout
+	cmd.Stderr = k.Stderr
+	cmd.Stdin = os.Stdin
+	return errors.Wrapf(cmd.Run(), "failed to run init script %s", scriptFile)
+}
+
+func promptYesNo(k *kong.Context, question string) (bool, error) {
```

Comment:
What happens if someone is calling `xpkg init` from a script? It would be nice to be able to provide a flag like `--run-init-script` that would skip this prompt if true.

Reply:
good idea, added